### PR TITLE
[install_script] Add option to send report to Datadog on error

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Add option to send error report to Datadog support when the install script finds an error. 
  The user is prompted for confirmation and for an email address for followup. 
  A report is then sent to Datadog including the installation logs.
- This tries to replicate the behavior of [the Agent v5 install script](https://github.com/DataDog/dd-agent/blob/2ce604ee0d7c70cdf85786deed689122b0e1957f/packaging/datadog-agent/source/setup_agent.sh#L187-L228).

### Motivation

- Make it easy for users to open a support case when they encounter an error.

### Additional Notes

- One can override the URL where failures are reported by using the `TESTING_REPORT_URL` environment variable.
- I did not add the changes to the Mac script since I believe the `on_error` function is currently not used.

### Describe your test plan

- Test normal installation on a supported platform. It should succeed as normally.
- Set `TESTING_REPORT_URL` to a local server that logs requests and cause an error in the script (e.g. by doing a parallel install in Ubuntu). Then:
    * If you answer 'No' to the prompt no request should be sent.
    * If you answer 'Yes' to the prompt then you should be asked for an email and a request should be sent to the server with the contents of the log file.
    * If you answer a different thing (not starting by y, Y, n or N) you should be prompted again

### CODE REVIEW

**Comment by mx-psi** [comment]
I am not sure if this should be on the changelog or not or how to go about it since it is an install script change. I have added the `changelog/no-changelog` label for now

File: cmd/agent/install_script.sh
Comment (albertvaka) [inline cmd/agent/install_script.sh:9]: I suggest we release this change right away (next week) after merging, so let's bump this version.

```suggestion
install_script_version=1.2.0
```

**Review by albertvaka — COMMENTED** [review]
There's a `releasenotes-installscript` directory we created not a long time ago, so we should start using it and create a CHANGELOG for the install_script. Can you do this as part of this PR?

**Comment by mx-psi** [comment]
Added release note but keeping the `changelog/no-changelog` since this checks for a release note in the Agent directory

**Review by albertvaka — APPROVED** [review]
LGTM, please trigger a release after merging.

**Pull request merged**
